File: Batalla_Poo_CPP/src/Tanque.cpp
#include "Tanque.h"

Tanque::Tanque()
{
    this->setNombre("Tanque");
}

/*
void Tanque::disparar(Participante enemigo)
{
    //enemigo.quitarVida(enemigo.danioParaTanque());
}
*/
int Tanque::danioParaTanque() const
{
    return 500;
}

int Tanque::danioParaAvion()
{
    return 600;
}

int Tanque::danioParaSoldado()
{
    return 250;
}

File: Batalla_Poo_Ruby/tanque.rb
require_relative "participante.rb"

class Tanque < Participante

    def getTipo
        return "Tanque"
    end

    def disparar(enemigo)
       enemigo.danio(enemigo.danioParaTanque())
    end

    def danioParaTanque()
        return 200
    end

    def danioParaSoldado()
        return 500
    end

    def danioParaAvion()
        return 650
    end

end